asv_bench/benchmarks/strftime.py

```diff
@@ -19,37 +19,68 @@ def setup(self, nobs):
                 "r": [np.random.uniform()] * nobs,
             }
         )
+        if tz_aware:
+            self.data["dt"] = self.data["dt"].dt.tz_localize("UTC")
+            self.data["d"] = self.data["d"].dt.tz_localize("UTC")
+
+        self.data["i"] = self.data["dt"]
+        self.data.set_index("i", inplace=True)
```

Comment:
The parameter tz_aware is being used to toggle between timezone-aware and naive timestamps. However, the logic for handling timezone-aware data in the setup method is somewhat repetitive and could benefit from encapsulation in a utility function.

Extract the logic for timezone localization into a helper function to improve readability and maintainability.
For example:
`
def localize_if_required(dataframe, tz_aware):
    if tz_aware:
        dataframe["dt"] = dataframe["dt"].dt.tz_localize("UTC")
        dataframe["d"] = dataframe["d"].dt.tz_localize("UTC")

`